### PR TITLE
Properly use ImportResult.sourceMapUrl for source map URLs

### PR DESCRIPTION
**Pull request by nex3**
This also uses data: URLs to refer to stylesheets from stdin in source
maps.

### CODE REVIEW

**Comment by nshahan** [comment]
This PR needs some cleanup first.

**Comment by nex3** [comment]
@nshahan Should be fixed now, although we'll see what the CI has to say about that...

**Comment by nex3** [comment]
@nshahan This is finally passing tests!

File: test/synchronize_test.dart [hunk 1]
```diff
@@ -10,14 +10,11 @@ import 'dart:io';
...
+
 void main() {
   test("synchronized files are up-to-date", () {
-    ({
-      'lib/src/visitor/async_evaluate.dart': 'lib/src/visitor/evaluate.dart',
-      'lib/src/async_environment.dart': 'lib/src/environment.dart',
-      'lib/src/async_import_cache.dart': 'lib/src/import_cache.dart'
-    })
-        .forEach((sourcePath, targetPath) {
+    synchronize.sources.forEach((sourcePath, targetPath) {
```
Comment (nshahan) [inline test/synchronize_test.dart]: Nice! This will avoid future pain.

**Pull request merged**